File: backend/app/schemas/ai.py
"""
AI service schemas for API requests and responses
"""
from pydantic import BaseModel, Field
from typing import List, Optional
from enum import Enum


class DifficultyLevel(str, Enum):
    """Issue difficulty levels"""
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"
    UNKNOWN = "unknown"


class LearningResource(BaseModel):
    """Learning resource recommendation"""
    title: str = Field(..., description="Resource title")
    url: str = Field(..., description="Resource URL")
    type: str = Field(..., description="Resource type (documentation, tutorial, video, etc.)")
    description: Optional[str] = Field(None, description="Brief description of the resource")


class RepositorySummaryRequest(BaseModel):
    """Request for repository summary generation"""
    repository_id: int = Field(..., description="Repository ID")
    force_regenerate: bool = Field(False, description="Force regeneration even if cached")


class RepositorySummaryResponse(BaseModel):
    """Response containing repository summary"""
    repository_id: int
    summary: str
    cached: bool = Field(..., description="Whether the summary was retrieved from cache")


class IssueExplanationRequest(BaseModel):
    """Request for issue explanation generation"""
    issue_id: int = Field(..., description="Issue ID")
    force_regenerate: bool = Field(False, description="Force regeneration even if cached")


class IssueExplanationResponse(BaseModel):
    """Response containing issue explanation"""
    issue_id: int
    explanation: str
    difficulty_level: DifficultyLevel
    learning_resources: List[LearningResource] = []
    cached: bool = Field(..., description="Whether the explanation was retrieved from cache")


class AIServiceError(BaseModel):
    """AI service error response"""
    error_code: str
    message: str
    details: Optional[dict] = None

File: frontend/src/types/ai.ts
export type DifficultyLevel = 'easy' | 'medium' | 'hard' | 'unknown'

export interface LearningResource {
  title: string
  url: string
  type: string
  description?: string
}

export interface RepositorySummaryResponse {
  repository_id: number
  summary: string
  cached: boolean
}

export interface IssueExplanationResponse {
  issue_id: number
  explanation: string
  difficulty_level: DifficultyLevel
  learning_resources: LearningResource[]
  cached: boolean
}

export interface ClaimResult {
  success: boolean
  message: string
  issue_id?: number
  claimed_at?: string
  claim_expires_at?: string
}

export interface ReleaseResult {
  success: boolean
  message: string
  issue_id?: number
}

export interface ExtensionResult {
  success: boolean
  message: string
  issue_id?: number
  new_expiration?: string
}
